src/variety/curvedata.js

```diff
@@ -0,0 +1,1272 @@
+    },
+
+    getBGCellColor : function(cell){
+        if(cell.qnum===-3) { return this.quescolor; }
+        if(cell.error) { return this.errbcolor1;}
+        return null;
+    }
+},
+
+"Graphic@curvedata-aux":{
+    gridcolor_type : "LIGHT",
+    linecolor: "black",
+
+    paint : function(){
+        this.drawBGCells();
+        this.drawDashedGrid();
+        this.drawLines();
+        this.drawChassis();
+    }
+},
+
+BoardExec:{
+    adjustBoardData : function(key,d){
+        if(key & this.TURNFLIP){
+            var trans = this.getTranslateBits(key);
+
+            var count = this.board.shapes.length;
+            for(var id = 0; id < count; id++) {
+                var data = this.board.shapes[id];
+                var w = data.w;
+                var len = data.bits.length;
+                var h = len/w;
+                data.buildBits();
+                var newBits = Array(len);
+
+                for(var b = 0; b < len; b++) {
+                    var bit = data.bits[b];
+                    if(bit in trans) {bit = trans[bit];}
+                    newBits[this.getTranslatePosition(key, w, h, b)] = bit;
+                }
+
+                if(key & this.TURN) { 
+                    data.w = h; 
+                    
+                    for(var y = 0; y < h; y++) {
+                        for(var x = 0; x < w; x++) {
+                            data.bits[x*h+y] = newBits[y*w+x];
+                        }
+                    }
+                } else {
+                    data.bits = newBits;
+                }
+                data.invalidate();
+            }
+
+            if(key === this.TURNR) {
+                this.adjustBoardData(this.FLIPX, d);
+            } else if(key === this.TURNL) {
+                this.adjustBoardData(this.FLIPY, d);
+            }
+        }
+    },
+
+    getTranslatePosition: function(key, w, h, id) {
+        var x = Math.floor(id%w);
+        var y = Math.floor(id/w);
+
+        switch(key){
+            case this.FLIPY:
+                return (h-(y+1))*w + x;
+            case this.FLIPX:
+                return y*w + (w-(x+1));
+            default:
+                return id;
+        }
+    },
+
+    getTranslateBits: function(key) {
+        var trans = {};
+        switch(key){
+            case this.FLIPY: 
+                trans={2:8, 3:9, 6:12, 7:13, 
+                    8:2, 9:3, 12:6, 13:7};
+                break;
+            case this.FLIPX:
+                trans={1:4, 3:6, 9:12, 11:14,
+                    4:1, 6:3, 12:9, 14:11};
+                break;
+            case this.TURNR:
+            case this.TURNL:
+                // Transpose
+                trans={1:2,2:1,4:8,5:10,6:9,7:11,8:4,9:6,10:5,11:7,13:14,14:13};
+                break;
+        }
+        return trans;
+    }
+},
+
+"Encode@curvedata":{
+    decodePzpr : function(type){
+        this.decodeClues();
+        var parts = this.outbstr.substr(1).split("/");
+        var i = 0;
```

Comment:
I'd prefer if this could be done without `i++`. I'm thinking something like

- a helper `decodeSlash()` that eats a slash (and if there's no slash we end the decode)
- decodeSlash, then if there's a "b" eat that and call `decodeBorder()` (would decodeBorder fail if the other parts are still at the end of `this.outbstr`?)
- decodeSlash, then split on slash, then e.g. iterate over `id` as we do currently but refer to `parts[3*id]`, `parts[3*id+1]`, `parts[3*id+2]`.